### PR TITLE
docs(cdn-caching): Generate cache keys and toggle CDN providers

### PR DESCRIPTION
**Pull request by Harshith46**
# Description

This PR handles the documentation for generating cache keys and how to toggle between CDN providers

### CODE REVIEW

File: server/caching.js [hunk 1]
```diff
@@ -1,4 +1,11 @@
 /**
+ * How Quintype handles caching
+ * - Whenever a collection/story/author is created or updated, the platform does cache/purging through CDN.
+ * - When a collection/story/author is created, the request body tag gets affected, then the frontend gets updated with the latest data with the help of cache-keys.
+ *
+ * How cache keys are created
+ * - Cache keys are generated in `quintype-node-framework` (server/caching.js). We have explained how cache keys are generated for each page type
```
Comment (Deokumarjnu) [inline server/caching.js:7]: we can give reference to `cache keys are generated` docs if we have.

File: server/handlers/cdn-caching.js [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+ *   How to toggle between cdn providers:
+ *   - If your app is cloned from malibu then, in `malibu/app/server/app.js` pass `cdnProvider` with required value to `isomorphicRoutes`(No need to pass if `cloudflare` is your cdn provider).
+ *
+ *   ```javascript
+ *   import { isomorphicRoutes } from "@quintype/framework/server/routes";
+ *   isomorphicRoutes(app, {
+ *   ...
+ *   ...
+ *   ...
+ *   cdnProvider: "YOUR_CND_PROVIDER_NAME",
```
Comment (Deokumarjnu) [inline server/handlers/cdn-caching.js:14]:
```suggestion
 *   cdnProvider: "YOUR_CDN_PROVIDER_NAME",
```

File: server/caching.js (hunk 1 above)
Comment (Harshith46) [reply server/caching.js:7]: it is in the same file

File: server/handlers/cdn-caching.js [hunk 3]
```diff
@@ -1,3 +1,31 @@
+/**
+ * How quintype handles toggle between different CDN providers
+ * - The default CDN provider is set as `Cloudflare` and can be toggled to the required CDN provider (for now we have support for `akamai`, in future we can support multiple CDN providers based on requirement).
```
Comment (nandap4790) [inline server/handlers/cdn-caching.js:3]: i think we can stop at for now we have support for `akamai`

File: server/handlers/cdn-caching.js (hunk 2 above)
Comment (Harshith46) [reply server/handlers/cdn-caching.js:14]: @Deokumarjnu ??

File: server/handlers/cdn-caching.js [hunk 4]
```diff
@@ -1,3 +1,31 @@
...
+ *
+ *   How to toggle between cdn providers:
+ *   - If your app is cloned from malibu then, in `malibu/app/server/app.js` pass `cdnProvider` with required value to `isomorphicRoutes`(No need to pass if `cloudflare` is your cdn provider).
```
Comment (nandap4790) [inline server/handlers/cdn-caching.js:6]: This again, you have to mention ppl to contact quintype before they do the change

**Review by nandap4790 — CHANGES_REQUESTED** [review]
Change the commit message to docs():--

File: server/handlers/cdn-caching.js (hunk 2 above)
Comment (Deokumarjnu) [reply server/handlers/cdn-caching.js:14]: typo, CND?

Comment (Harshith46) [reply server/handlers/cdn-caching.js]: oh ok

File: server/caching.js [hunk 5]
```diff
@@ -1,4 +1,11 @@
 /**
+ * How Quintype handles caching
+ * - Whenever a collection/story/author is created or updated, the platform does cache/purging through CDN.
+ * - When a collection/story/author is created, the request body tag gets affected, then the frontend gets updated with the latest data with the help of cache-keys.
```
Comment (rashmirmittal) [inline server/caching.js:4]: what is request body tag?

File: server/handlers/cdn-caching.js [hunk 6]
```diff
@@ -1,3 +1,33 @@
+/**
+ * How quintype handles toggle between different CDN providers
+ * - For now we have support for `Cloudflare` and `akamai`.
```
Comment (rashmirmittal) [inline server/handlers/cdn-caching.js:3]: We also support Fastly

File: server/handlers/cdn-caching.js [hunk 7]
```diff
@@ -1,3 +1,33 @@
...
+ *   cdnProvider: "YOUR_CDN_PROVIDER_NAME",
+ *   });
+ *   ```
+ *   - In `quintype-node-framework`, the functionality to set the required response headers is written in file: `server/handlers/cdn-caching.js`
+ *   - If the cdn provider is `akamai`, then we are updating the response headers as below
+ *   ```javascript
+ *   res.setHeader("Edge-Control", "private,no-cache,no-store,max-age=0");
+ *   res.setHeader("Edge-Control",`public,maxage=${sMaxAge},stale-while-revalidate=1000,stale-if-error=14400`);
+ *   res.setHeader("Edge-Cache-Tag", _(cacheKeys).uniq().join(","));
+ *   res.setHeader("Edge-Control","public,maxage=60,stale-while-revalidate=150,stale-if-error=3600");
```
Comment (rashmirmittal) [inline server/handlers/cdn-caching.js:25]: If someone is not using our fwk, like esakal, we should give examples of setting CF and Fastly headers, Fastly may be used as backup CDN

**Pull request closed without merging**